src/Valet/Interfaces/IConfigurationService.cs

```diff
@@ -1,18 +1,15 @@
+using System.Collections.Immutable;
+
 namespace Valet.Interfaces;
 
 public interface IConfigurationService
 {
-    Task<Dictionary<string, string>> ReadCurrentVariablesAsync(string filePath = ".env.local");
-    Dictionary<string, string> GetUserInput();
-    Task WriteVariablesAsync(Dictionary<string, string> variables, string filePath = ".env.local");
+    Task<ImmutableDictionary<string, string>> ReadCurrentVariablesAsync(string filePath = ".env.local");
+    ImmutableDictionary<string, string> GetUserInput();
+    Task WriteVariablesAsync(ImmutableDictionary<string, string> variables, string filePath = ".env.local");
 
-    Dictionary<string, string> MergeVariables(Dictionary<string, string> currentVariables, Dictionary<string, string> newVariables)
+    ImmutableDictionary<string, string> MergeVariables(ImmutableDictionary<string, string> currentVariables, ImmutableDictionary<string, string> newVariables)
     {
-        foreach (var variable in newVariables)
-        {
-            currentVariables[variable.Key] = variable.Value;
-        }
-
-        return currentVariables;
+        return currentVariables.SetItems(newVariables);
     }
```

Comment:
The safety of the change here is covered by an existing unit test `MergeVariables_OverwritesExisting`